app/utils/entity_utils.rb

```diff
@@ -36,81 +36,79 @@ def model_to_hash(model)
-    date: -> (_, v, field) {
+    date: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Date))
-        "#{field}: Value must be a Date. Was: #{v} (#{v.class.name})."
+        {code: :date, msg: "Value must be a Date. Was: #{v} (#{v.class.name})." }
       end
     },
-    fixnum: -> (_, v, field) {
+    fixnum: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Fixnum))
-        "#{field}: Value must be a Fixnum. Was: #{v} (#{v.class.name})."
+        {code: :fixnum, msg: "Value must be a Fixnum. Was: #{v} (#{v.class.name})." }
       end
     },
-    symbol: -> (_, v, field) {
+    symbol: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Symbol))
-        "#{field}: Value must be a Symbol. Was: #{v} (#{v.class.name})."
+        {code: :symbol, msg: "Value must be a Symbol. Was: #{v} (#{v.class.name})." }
       end
     },
-    hash: -> (_, v, field) {
+    hash: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Hash))
-        "#{field}: Value must be a Hash. Was: #{v} (#{v.class.name})."
+        {code: :hash, msg: "Value must be a Hash. Was: #{v} (#{v.class.name})." }
       end
     },
-    callable: -> (_, v, field) {
+    callable: -> (_, v, _) {
       unless (v.nil? || v.respond_to?(:call))
-        "#{field}: Value must respond to :call, i.e. be a Method or a Proc (lambda, block, etc.)."
+        {code: :callable, msg: "Value must respond to :call, i.e. be a Method or a Proc (lambda, block, etc.)." }
       end
     },
-    enumerable: -> (_, v, field) {
+    enumerable: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Enumerable))
-        "#{field}: Value must be an Enumerable. Was: #{v}."
+        {code: :enumerable, msg: "Value must be an Enumerable. Was: #{v}." }
       end
     },
-    array: -> (_, v, field) {
+    array: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Array))
-        "#{field}: Value must be an Array. Was: #{v}."
+        {code: :array, msg: "Value must be an Array. Was: #{v}." }
       end
     },
-    set: -> (_, v, field) {
+    set: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Set))
-        "#{field}: Value must be a Set. Was: #{v} (#{v.class.name})."
+        {code: :set, msg: "Value must be a Set. Was: #{v} (#{v.class.name})." }
       end
     },
-    money: -> (_, v, field) {
+    money: -> (_, v, _) {
       unless (v.nil? || v.is_a?(Money))
-        "#{field}: Value must be a Money. Was: #{v}."
+        {code: :money, msg: "Value must be a Money. Was: #{v}." }
       end
     },
-    bool: -> (_, v, field) {
+    bool: -> (_, v, _) {
       unless (v.nil? || v == true || v == false)
-        "#{field}: Value must be boolean true or false. Was: #{v} (#{v.class.name})."
+        {code: :bool, msg: "Value must be boolean true or false. Was: #{v} (#{v.class.name})." }
       end
     },
-    validate_with: -> (validator, v, field) {
-      unless (validator.call(v))
-        "#{field}: Custom validation failed. Was: #{v}."
-      end
+    validate_with: -> (validator, v, _) {
+      validator.call(v)
```

Comment:
The validator return value type change is not backwards compatible. If there's any existing code that uses `validate_with`it will break. Do we have these?